### PR TITLE
fix(logging): improve node core's debug logging

### PR DESCRIPTION
**Pull request by padamstx**
This commit adds some debug logging in various components.

Edit: it also includes a more custom logger for http request/response messages

### CODE REVIEW

File: lib/request-wrapper.ts [hunk 1]
```diff
@@ -391,16 +474,52 @@ export class RequestWrapper {
...
+        if (error.config.url) {
+          details.push(`url=${error.config.url}`);
+        }
+      }
+
+      logger.debug(`Considering retry attempt; ${details.join(', ')}`);
+    }
+
+    // Delegate to the default function defined by retry-axios.
+    const shouldRetry = rax.shouldRetryRequest(error);
```
Comment (padamstx) [inline lib/request-wrapper.ts:516]: I chickened out and just made this function a wrapper around the default implementation found in retry-axios.  Later, if we decide to make the retry criteria 100% consistent across the cores, we could change this to perform our custom logic here instead.

File: lib/private-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+];
+
+const redactedTokens = redactedKeywords.join('|');
+
+// Pre-compiled regular expressions used by redactSecrets().
+const reAuthHeader = new RegExp(`^(Authorization|X-Auth\\S*): .*$`, 'gim');
+const rePropertySetting = new RegExp(`(${redactedTokens})=[^&]*(&|$)`, 'gi');
+const reJsonField = new RegExp(`"([^"]*(${redactedTokens})[^"_]*)":\\s*"[^\\,]*"`, 'gi');
+
+// RedactSecrets() returns the input string with secrets redacted.
```
Comment (pyrooka) [inline lib/private-helpers.ts:49]:
```suggestion
// redactSecrets() returns the input string with secrets redacted.
```
👀

File: lib/request-wrapper.ts [hunk 3]
```diff
@@ -367,7 +449,8 @@ export class RequestWrapper {
       instance: axiosInstance,
       backoffType: 'exponential',
       checkRetryAfter: true, // use Retry-After header first
-      maxRetryDelay: 30 * 1000, // default to 30 sec max delay
+      maxRetryDelay: 30 * 1000, // default to 30 sec max delay,
```
Comment (pyrooka) [inline lib/request-wrapper.ts:452]:
```suggestion
      maxRetryDelay: 30 * 1000, // default to 30 sec max delay
```
Feel free to ignore this, but I don't think that extra comma at the end of the comment is needed. :)

File: test/unit/vpc-instance-token-manager.test.js [hunk 4]
```diff
@@ -39,7 +39,11 @@ const IAM_PROFILE_ID = 'some-id';
 const EXPIRATION_WINDOW = 10;
 
 describe('VPC Instance Token Manager', () => {
-  const sendRequestMock = jest.fn().mockImplementation(() => ({ result: { access_token: TOKEN } }));
+  const sendRequestMock = jest.fn(); // .mockImplementation(() => ({ result: { access_token: TOKEN } }));
```
Comment (pyrooka) [inline test/unit/vpc-instance-token-manager.test.js:42]:
```suggestion
  const sendRequestMock = jest.fn();
```
I guess this comment can be removed, right?

**Review by pyrooka — APPROVED** [review]
Looks good. I really like the request-wrapper related changes. :)

File: test/unit/redact-secrets.test.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+ */
+
+const { redactSecrets } = require('../../dist/lib/private-helpers');
+
+describe('redactSecrets()', () => {
+  it('misc. tests', async () => {
+    const inputStrings = [
+      'Authorization: Bearer secret',
+      'Authorization: secret\nX-Author: secret',
+      'apikey=81KiI5Zm2kjOWnrSglhtnDJn3u0kfv&grant_type=apikey&response_type=cloud_iam',
```
Comment (dpopp07) [inline test/unit/redact-secrets.test.js:24]: I'm guessing detect secrets will have a field day with this 😂 and it looks like you haven't updated the baseline yet - just a reminder

**Review by dpopp07 — APPROVED** [review]
Looks good!

File: lib/request-wrapper.ts (hunk 3 above)
Comment (padamstx) [reply lib/request-wrapper.ts:452]: fixed

File: test/unit/vpc-instance-token-manager.test.js (hunk 4 above)
Comment (padamstx) [reply test/unit/vpc-instance-token-manager.test.js:42]: fixed

File: test/unit/redact-secrets.test.js (hunk 5 above)
Comment (padamstx) [reply test/unit/redact-secrets.test.js:24]: good catch!

**Pull request merged**